cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -1382,6 +1382,25 @@ TEST_F(TestAzuriteFileSystem, DeleteDirContentsFailureNonexistent) {
   this->TestDeleteDirContentsFailureNonexistent();
 }
 
+TEST_F(TestAzuriteFileSystem, DeleteFileSuccess) {
+  CreateFile(fs(), "abc", "data");
+  arrow::fs::AssertFileInfo(fs(), "abc", FileType::File);
+  ASSERT_OK(fs()->DeleteFile("abc"));
+  arrow::fs::AssertFileInfo(fs(), "abc", FileType::NotFound);
+}
+
+TEST_F(TestAzuriteFileSystem, DeleteFileFailureNonexistent) {
+  arrow::fs::AssertFileInfo(fs(), "abc", FileType::NotFound);
+  ASSERT_RAISES(IOError, fs()->DeleteFile("abc"));
```

Comment:
How about simplifying this?

```suggestion
  ASSERT_RAISES(IOError, fs()->DeleteFile("nonexistent"));
```

Reply:
OK done.